app/src/main/java/gov/hhs/cdc/trustedintermediary/external/jjwt/JjwtEngine.java

```diff
@@ -0,0 +1,104 @@
+package gov.hhs.cdc.trustedintermediary.external.jjwt;
+
+import gov.hhs.cdc.trustedintermediary.wrappers.AuthEngine;
+import io.jsonwebtoken.*;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Date;
+import java.util.UUID;
+import org.jetbrains.annotations.NotNull;
+
+public class JjwtEngine implements AuthEngine {
+
+    private static final JjwtEngine INSTANCE = new JjwtEngine();
+
+    private JjwtEngine() {}
+    ;
+
+    public static JjwtEngine getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    @NotNull
+    public String generateSenderToken(
+            @NotNull String sender,
+            @NotNull String baseUrl,
+            @NotNull String pemKey,
+            @NotNull String keyId,
+            int expirationSecondsfromNow)
+            throws InvalidKeySpecException, NoSuchAlgorithmException {
+
+        JwtBuilder jwsObj =
+                Jwts.builder()
+                        .setHeaderParam("kid", keyId) // flexion.etor-service-sender
+                        .setHeaderParam("typ", "JWT")
+                        .setIssuer(sender) // flexion.etor-service-sender
+                        .setSubject(sender) // flexion.etor-service-sender
+                        .setAudience(baseUrl) // "https://staging.prime.cdc.gov/api/token"
+                        .setExpiration(
+                                new Date(
+                                        System.currentTimeMillis()
+                                                + (long) (expirationSecondsfromNow * 1000)))
```

Comment:
You can do away with the explicit cast by making making the `1000` a long time by appending an `L`.

```suggestion
                                                + (expirationSecondsfromNow * 1000L)))
```

Reply:
Good catch!